### ISSUE TITLE
Please upload the pretrained HiDDeN decoder weights.

### ISSUE THREAD

**Issue by DI-LEE** [facebookresearch/stable_signature#5]
Hello, I'm DI in Korea univ.
I'm very impressed with your research.
So I want to reproduce various bits of it, but I failed to pretrain 32 bits HiDDeN decoder.
Please upload the pretrained 32 bits decoder's weights or run code please.

Thank you.
Best regards
DI Lee

**Comment by pierrefdz** [comment]
Hi,
The Hidden models are already uploaded (see the readme.md).
The code to run watermark networks is in https://github.com/facebookresearch/stable_signature/tree/main/hidden.
Tell me if I'm missing smthing.

**Comment by DI-LEE** [comment]
Thank you for fast reply.
I'm reproducing 32 bits HiDDeN decoder but Its bit accuracy only goes up to 87%.
I'm trying to get 100%, 
Can I have access to the decoder weights used in your code?

**Comment by pierrefdz** [comment]
I havent used 32bits decoder. 
The 48bits one is given in the reproduction: https://github.com/facebookresearch/stable_signature/tree/main/hidden/ckpts
Or in https://github.com/facebookresearch/stable_signature/tree/main#watermark-models

What is the decoder you are referring to?
You can also share the parameters used for your training (the optimization can be a bit unstable depending on them)

**Comment by pierrefdz** [comment]
What is your setup? Can you share the logs of your run?

This command should work directly if you have the same setup (and replacing 48 by 32).
https://github.com/facebookresearch/stable_signature/tree/main/hidden#example

If you don't have the same setup (for instance number of gpus), you can try reducing the lr to lower values like 1e-3.

**Comment by pierrefdz** [comment]
Thank you, I'll try to have a look at it in the following days.

**Comment by pierrefdz** [comment]
Hi DI, 
I've run the code on my side and accuracy gets to 1.00, here are the logs:
https://drive.google.com/file/d/1qG21QKmVkikQlQ4c2mzqDIsLNe04bQqp/view?usp=drive_link

Could you share with me the logs of your console s.t. I can see if there is an issue?
Otherwise there might be a difference either on the dataset, or in the code I run (but I don't think so since the code I've used is the one I pushed here...)

**Comment by DI-LEE** [comment]
I'm very appreciate it.
But I can't access your drive, please give me access right.

**Comment by pierrefdz** [comment]
can you ask for it? otherwise I need an email to give you access rights

**Comment by DI-LEE** [comment]
My email address is `dongeen1@gmail.com`.
And my log is below.

[log.txt](https://github.com/facebookresearch/stable_signature/files/13238405/log.txt)

**Comment by DI-LEE** [comment]
I'm really sorry to ask, but could you possibly upload the weight file for the 32bit pretrained model to the drive?

**Comment by pierrefdz** [comment]
Our logs do not share the same lr, so I think the issue comes from here (my guess is that your world size is different in your distributed setup, so the lr is scaled differently - I have not enough experiments on this and did the same scaling as image recognition, which might not be adapted).

You can try removing this line https://github.com/facebookresearch/stable_signature/blob/main/hidden/main.py#L218 and fix the lr to 0.005 instead of 2e-2.

I won't be able to share the networks (it would need to go through internal review), sorry for that.

**Comment by pierrefdz** [comment]
Closing since no activity, feel free to re-open!

**Comment by JosTheBossX** [comment]
Hello, I just checked the output logs from https://drive.google.com/file/d/1qG21QKmVkikQlQ4c2mzqDIsLNe04bQqp/view?usp=drive_link

So for me, I tried to train 8bit decoder and 32bit with the same parameters that were given to replicate (just changed the --num_bits to 8 and 32, and also I am training it on 2 GPUs)

So from my logs I see that I am getting 0.99 bit ACC for train and val both, but when I try to use the checkpoints of my saved model using the demo.ipynb file, I do not get 100 bit ACC on both 8 or 32 bit experiments. For both of them I get around 87.

I am gonna try to do this https://github.com/facebookresearch/stable_signature/issues/5#issuecomment-1790640271
and see if it helps me.

But I would also like to ask one more thing that, the original Hidden network states that it was trained on 10k train + 1k val dataset, but I see that in this implementation it isn't mentioned how much of the data we use from the 2014 coco dataset for train and val, do we use all the 80k images to train from train2014?
